### PR TITLE
[MENU/ JOYSTICK] adding fuzz value array

### PR DESCRIPTION
**Pull request by brandonrc**
Replacing the single fuzz value for an array of fuzz values. 

Jumper did not do a linear resistor ladder so the values are different based on position. The solution is to check each value and see what is the smallest distance between the next value and divide that in half. This will avoid collisions. 

Tested on ghost tx and ghost tx lite.

### CODE REVIEW

**Review by CapnBry — CHANGES_REQUESTED** [review]
I've listed a lot of things as "should be static" but I'm not sure why these aren't private functions on the FiveWayButton class? And can you remove these unused defines?
```
#define UP    0;
#define DOWN  1;
#define LEFT  2;
#define RIGHT 3;
#define ENTER 4;
#define IDLE  5;
```

File: src/lib/SCREEN/FiveWayButton/FiveWayButton.cpp [hunk 1]
```diff
@@ -21,21 +21,70 @@ int key = INPUT_KEY_NO_PRESS;
...
+void findMinDiff()
+{
+    int n = 6; //Number of positions up, down, left, right, enter, idle
+    
+    // First for loop is stepping through the the real JOY_ADC_VALUES
+    for (int i=0; i < n; i++)
+        // Second for loop is stepping through the sorted values to find its min
+        // distance between (curr - 1) and (curr + 1)
+        for (int j = 0; j < n; j++)
+            if(joyAdcValues[i] == sortedJoyAdcValues [j] ){
```
[the remark this answers is not in the document]
Comment (brandonrc) [reply src/lib/SCREEN/FiveWayButton/FiveWayButton.cpp:68]: That is my bad.. it was a quick proof of concept this morning before I ran to work :)

File: src/lib/SCREEN/FiveWayButton/FiveWayButton.cpp [hunk 2]
```diff
@@ -21,21 +21,70 @@ int key = INPUT_KEY_NO_PRESS;
 
 static int16_t joyAdcValues[] = JOY_ADC_VALUES;
 
-// Returns minimum difference between any pair
-int findMinDiff(int16_t arr[], int n)
+int16_t fuzzValues[] = {0,0,0,0,0,0};
```
[the remark this answers is not in the document]
Comment (brandonrc) [reply src/lib/SCREEN/FiveWayButton/FiveWayButton.cpp]: What do you mean by `helpers.h`. 

Like move some of the code to a the helpers.h file and include that here?

**Comment by CapnBry** [comment]
* I've moved all the code into the FiveWayButton class instead of being in the global namespace
* Replaced sorting the array to calculate the fuzz values with a simple search for the closest neighbor
* Replaced `handle() + getKeyState()` with just a single `update()` that returns keyState
* Added compiler errors for if GPIO_PIN_JOYSTICK is defined and JOY_ADC_VALUES is not and vice versa
* Changed KEY_DEBOUNCE_TIME to actually debounce the press event. I'm not sure what the old code was doing, preventing a second keypress within a certain amount of time?

Reduced code size by 440 bytes and RAM by 16 bytes for analog 5ways.

Tested on Namimno Flash OLED and Axis Thor.

**Review by CapnBry — APPROVED** [review]

**Review by schugabe — APPROVED** [review]
Tested on the aion nano module that was the reason for the change.

**Comment by brandonrc** [comment]
Tested on (and works):
Ghost TX Full
Ghost TX Lite
BetaFPV Micro 2.4 1000mW
NamimnoRC Flash 2.4 OLED

**Pull request merged**